src/olympia/translations/fields.py

```diff
@@ -286,42 +276,6 @@ def adapt(cls, opts=None):
         return type('Trans%s' % cls.__name__, (_TransField, cls), opts)
 
 
-# Subclass list so that isinstance(list) in Django works.
-class LocaleList(dict):
-    """
-    List-like objects that maps list elements to a locale.
-
-    >>> LocaleList([1, 2], 'en')
-    [1, 2]
-    ['en', 'en']
-
-    This is useful for validation error lists where we want to associate an
-    error with a locale.
-    """
-
-    def __init__(self, seq=None, locale=None):
-        self.seq, self.locales = [], []
-        if seq:
-            assert seq and locale
-            self.extend(seq, locale)
-
-    def __iter__(self):
-        return iter(self.zip())
-
-    def extend(self, seq, locale):
-        self.seq.extend(seq)
-        self.locales.extend([locale] * len(seq))
-
-    def __nonzero__(self):
-        return bool(self.seq)
-
-    def __contains__(self, item):
-        return item in self.seq
-
-    def zip(self):
-        return zip(self.locales, self.seq)
```

Comment:
I'd see this related to the 1.11 work but afaik `form.add_error` already exists in 1.8 so it could be worth to explore uplifting this too. I'd rather not spend too much time on this though, stupidly complex stuff our form error handling with translations :slightly_frowning_face: